daikon-tql/daikon-tql-dsel/src/main/java/org/talend/tqldsel/dseltotql/DselToTqlVisitor.java

```diff
@@ -315,6 +325,34 @@ private TqlElement visitFakeEmptyFunction() {
         return buildNewAST(new FieldIsEmptyExpression(new AllFields()));
     }
 
+    private TqlElement visitCompliesFunction(ELNode elNode) {
+        final TqlElement field = elNode.getChild(0).accept(this);
+        final ELNode fieldTypeNode = elNode.getChild(1);
+
+        if (!ELNodeType.STRING_LITERAL.equals(fieldTypeNode.getType())) {
+            throw new IllegalArgumentException(
+                    "The second parameter is not a STRING literal, whereas the complies function can only accept a STRING literal as second parameter");
+        }
+
+        final LiteralValue fieldTypeLiteralValue = (LiteralValue) visitLiteral(fieldTypeNode);
+
+        return buildNewAST(new FieldCompliesPattern(field, fieldTypeLiteralValue.getValue()));
+    }
+
+    private TqlElement visitWordCompliesFunction(ELNode elNode) {
+        final TqlElement field = elNode.getChild(0).accept(this);
+        final ELNode fieldTypeNode = elNode.getChild(1);
+
+        if (!ELNodeType.STRING_LITERAL.equals(fieldTypeNode.getType())) {
+            throw new IllegalArgumentException(
+                    "The second parameter is not a STRING literal, whereas the complies function can only accept a STRING literal as second parameter");
```

Comment:
fix exception message
```suggestion
                    "The second parameter is not a STRING literal, whereas the wordComplies function can only accept a STRING literal as second parameter");
```

Reply:
Thanks!